api/src/main/java/io/hyperfoil/api/config/BuilderBase.java

```diff
@@ -86,7 +86,7 @@ private void tryPrepare(Class<?> clz, String name, Class<?> type, Object value)
    default S copy(Object newParent) {
       if (getClass().isSynthetic()) {
          // This is most likely a lambda supplier of the instance (which should be immutable anyway)
-         assert getClass().getSimpleName().contains("$$Lambda$");
+         assert getClass().getSimpleName().contains("$$Lambda");
```

Comment:
Looks like something changed on how the simple name is composed in case of lambdas, still need to confirm this

Reply:
I can confirm this changed happened in the JDK 21:
* Related JDK issue: https://bugs.openjdk.org/browse/JDK-8292914
* Commit: https://github.com/openjdk/jdk/commit/b527edd3388ad6a0d44a291983b08b2b5c023f8f

```
        }
-        return name.replace('.', '/') + "$$Lambda$" + counter.incrementAndGet();
+        return name.replace('.', '/') + "$$Lambda";
    }
```

Reply:
So we don't need to have a Runtime version check to pick the right way to name them?
Or apply some static check on a know lambda name and decide the strategy?
From my understanding relying on how lambdas are named can be...brittle

Reply:
Regardless, well done to have found the issue eh : ❤️